assets/src/components/propertiesPanel/minischedule.tsx

```diff
@@ -1,33 +1,68 @@
-import React from "react"
+import React, { ReactElement } from "react"
 import {
   useMinischeduleBlock,
   useMinischeduleRun,
 } from "../../hooks/useMinischedule"
-import { Block, Run } from "../../minischedule"
+import { Block, Break, Piece, Run, Trip } from "../../minischedule"
 import { TripId } from "../../schedule"
 import Loading from "../loading"
 
 export interface Props {
   activeTripId: TripId
 }
-export const MinischeduleRun = ({ activeTripId }: Props) => {
+export const MinischeduleRun = ({ activeTripId }: Props): ReactElement => {
   const run: Run | null | undefined = useMinischeduleRun(activeTripId)
   if (run === undefined) {
     return <Loading />
   } else if (run === null) {
     return <>No run found</>
   } else {
-    return <div>{JSON.stringify(run)}</div>
+    return (
+      <>
+        {run.activities.map((activity) =>
+          isPiece(activity) ? (
+            <Piece piece={activity} key={activity.start.time} />
+          ) : (
+            <Break break={activity} key={activity.startTime} />
+          )
+        )}
+      </>
+    )
   }
 }
 
-export const MinischeduleBlock = ({ activeTripId }: Props) => {
+export const MinischeduleBlock = ({ activeTripId }: Props): ReactElement => {
   const block: Block | null | undefined = useMinischeduleBlock(activeTripId)
   if (block === undefined) {
     return <Loading />
   } else if (block === null) {
     return <>No block found</>
   } else {
-    return <div>{JSON.stringify(block)}</div>
+    return (
+      <>
+        {block.pieces.map((piece) => (
+          <Piece piece={piece} key={piece.start.time} />
+        ))}
+      </>
+    )
   }
 }
+
+const Break = ({ break: breakk }: { break: Break }) => (
```

Comment:
Typo: `breakk`

Reply:
That's actually intentional. `break` is a keyword in js, so you can't name a variable that. Since "Break" is such a specific concept, I figured modifying the name was better than trying to find a synonym, but if you have other ideas on how to handle this, I'm open to it.

Reply:
Oh, ha! Makes sense.